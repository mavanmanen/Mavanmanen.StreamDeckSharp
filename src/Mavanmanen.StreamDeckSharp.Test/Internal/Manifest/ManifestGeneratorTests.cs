using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Mavanmanen.StreamDeckSharp.Attributes;
using Mavanmanen.StreamDeckSharp.Enum;
using Mavanmanen.StreamDeckSharp.Internal.Definitions;
using Mavanmanen.StreamDeckSharp.Internal.Manifest;
using Xunit;
using Xunit.Abstractions;

namespace Mavanmanen.StreamDeckSharp.Test.Internal.Manifest
{
    [Trait("Category", "Unit Tests")]
    [UseReporter(typeof(DiffReporter))]
    public class ManifestGeneratorTests : XunitApprovalBase
    {
        public ManifestGeneratorTests(ITestOutputHelper output) : base(output)
        {
        }

        [Fact]
        public void GenerateManifest_GeneratesCorrectManifest()
        {
            // Arrange
            var plugin = new PluginDefinition(typeof(Plugin));
            var actions = new List<ActionDefinition>
            {
                new ActionDefinition(typeof(Action))
            };
            var sut = new ManifestGenerator(plugin, actions);

            // Act
            string manifestJson = sut.GenerateManifest();

            // Assert
            Approvals.Verify(manifestJson);
        }
    }

    [StreamDeckPlugin("pluginName", "icon", "author", "description", "1.0", "url", "category", "categoryIcon", defaultWindowSize: "560,120")]
    [StreamDeckMinimumOsVersion("windowsMinimumVersion", "macMinimumVersion")]
    [StreamDeckApplicationsToMonitor(new []{ "notepad.exe" }, new []{ "notepad" })]
    [StreamDeckProfile("profile", DeviceType.StreamDeck, true, true)]
    public class Plugin : StreamDeckPlugin
    {

    }

    [StreamDeckAction("name", "icon", "tooltip", supportedInMultiActions: true, visibleInActionsList: true)]
    [StreamDeckActionState("image", "state1", "multiActionImage", "state1", false, "#000000", TitleAlignment.Middle, FontFamily.Arial, FontStyle.Bold, "12", false)]
    [StreamDeckActionState("image", "state2", "multiActionImage", "state2", false, "#000000", TitleAlignment.Middle, FontFamily.Arial, FontStyle.Bold, "12", false)]
    public class Action : StreamDeckAction
    {

    }
}
