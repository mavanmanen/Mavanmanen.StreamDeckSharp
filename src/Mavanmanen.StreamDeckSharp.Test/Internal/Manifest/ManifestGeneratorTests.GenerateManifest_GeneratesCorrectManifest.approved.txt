{
  "Actions": [
    {
      "Icon": "icon",
      "Name": "name",
      "States": [
        {
          "Image": "image",
          "MultiActionImage": "multiActionImage",
          "Name": "state1",
          "Title": "state1",
          "ShowTitle": false,
          "TitleColor": "#000000",
          "TitleAlignment": "Middle",
          "FontFamily": "Arial",
          "FontStyle": "Bold",
          "FontSize": 12,
          "FontUnderline": false
        },
        {
          "Image": "image",
          "MultiActionImage": "multiActionImage",
          "Name": "state2",
          "Title": "state2",
          "ShowTitle": false,
          "TitleColor": "#000000",
          "TitleAlignment": "Middle",
          "FontFamily": "Arial",
          "FontStyle": "Bold",
          "FontSize": 12,
          "FontUnderline": false
        }
      ],
      "SupportedInMultiActions": true,
      "Tooltip": "tooltip",
      "UUID": "mavanmanen.streamdecksharp.test.internal.manifest.name",
      "VisibleInActionsList": true
    }
  ],
  "Author": "author",
  "Category": "category",
  "CategoryIcon": "categoryIcon",
  "CodePathMac": "mavanmanen.streamdecksharp.test.internal.manifest",
  "CodePathWin": "mavanmanen.streamdecksharp.test.internal.manifest.exe",
  "Description": "description",
  "Icon": "icon",
  "Name": "pluginName",
  "Profiles": [
    {
      "Name": "profile",
      "DeviceType": "kESDSDKDeviceType_StreamDeck",
      "ReadOnly": true,
      "DontAutoSwitchWhenInstalled": true
    }
  ],
  "DefaultWindowSize": [
    560,
    120
  ],
  "URL": "url",
  "Version": "1.0",
  "SDKVersion": 2,
  "OS": [
    {
      "Platform": "Windows",
      "MinimumVersion": "windowsMinimumVersion"
    },
    {
      "Platform": "Mac",
      "MinimumVersion": "macMinimumVersion"
    }
  ],
  "Software": {
    "MinimumVersion": "4.1"
  },
  "ApplicationsToMonitor": {
    "mac": [
      "notepad"
    ],
    "windows": [
      "notepad.exe"
    ]
  }
}